drones/connection/connector.py

```diff
@@ -0,0 +1,153 @@
+        Receive state UDP datagrams from Tello, log socket error, close socket on exceptions.
+    send_command(command: str)
+        Sends command to Tello
+    """
+
+    def __init__(self):
+        config = configparser.ConfigParser()
+        config.read("connection/config.ini")
+        config = config["NETWORK"]
+        self.address_response = (config["host"], config.getint("port_response"))
+        self.address_state = (config["host"], config.getint("port_state"))
+        self.stream_address = config["stream_address"]
+        self.tello_address = (config["tello_address"], config.getint("tello_port"))
+        self.state_byte_size = config.getint("state_byte_size")
+        self.response_byte_size = config.getint("response_byte_size")
+        self.socket_receive_response = None
+        self.socket_receive_state = None
+        self.tello_connected = False
+        self.response_receiving_thread = None
+        self.state_receiving_thread = None
+        self.state = None
+        self.response = None
+        self.last_command = None
+
+    def connect(self) -> None:
+        """Establish connection to Tello and enable SDK."""
+
+        try:
+            self.socket_receive_response = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+            self.socket_receive_response.bind(self.address_response)
+            self.socket_receive_state = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+            self.socket_receive_state.bind(self.address_state)
+        except (OSError, socket.herror, socket.gaierror) as err:
+            log.error(f"Error {err.errno}: {err.strerror}")
+        else:
+            self.tello_connected = True
+            self.response_receiving_thread = threading.Thread(target=self.receive_response)
+            self.response_receiving_thread.start()
+            self.state_receiving_thread = threading.Thread(target=self.receive_state)
+            self.state_receiving_thread.start()
+            self.send_command("command")
+            log.info("Connection established.")
+
+    def disconnect(self) -> None:
+        """Close sockets"""
+
+        if self.tello_connected:
+            self.socket_receive_response.close()
+            self.socket_receive_state.close()
+            self.tello_connected = False
+            log.info("Connection closed.")
+        else:
+            log.info("Connection is closed already.")
+
+    def receive_response(self) -> None:
+        """Receive command response UDP datagrams from Tello, log socket error, close socket on exceptions."""
+
+        # TODO
+        # Await response with timeout
+
+        while True:
+            try:
+                data, server = self.socket_receive_response.recvfrom(self.response_byte_size)
+                self.response = data.decode(encoding="utf-8")
+                log.info(self.response + "\n")
+
+            except OSError as err:
+                log.error(err)
+                break
+
+        self.tello_connected = False
```

Comment:
Why are we setting `self.tello_connected` to `False` here? Is it because catching `OSError` assumes we lost connection? Shouldn't we execute `self.disconnect()` instead? I think a logging message about losing connection would be nice here.

Reply:
You're right, my bad. Flagging Tello as disconnected would cause throwing and exception while binding sockets again.